server/controllers/request.js

```diff
@@ -0,0 +1,51 @@
+const Request = require("../models/Request");
+const asyncHandler = require("express-async-handler")
+
+// @route POST /requests/create
+// @desc Create new request
+// @access Public
+exports.createRequest = asyncHandler(async (req, res, next) => {
+  const request = await Request.create(req.body);
+
+  if (request) {
+    res.status(200).json({
+      success: {
+        request: {
+          ...request
+        }
+      }
+    })
+  }
+});
+
+// @route PUT /requests/edit/:requestId
+// @desc edit request
+// @access Public
+exports.editRequest = asyncHandler(async (req, res, next) => {
+  const request = await Request.findById(req.params.requestId)
+  
+  if (!request) {
+    res.status(404);
+    throw new Error("Request doesn't exist");
+  }
+  request.set(req.body);
+  const updatedRequest = await request.save();
+  res.status(200).json({
+    success: {
+      request: updatedRequest
+    },
+  });
+});
+
+// @route GET /requests/?userType=
+// @desc get requests of logged in user
+// @access Private
+exports.getUserRequests = asyncHandler(async (req, res, next) => {
+  const requests = await Request.where(req.query.userType, req.body.user.id);
```

Comment:
Let's refrain from using the req.body in a GET request. A payload within a GET request message has no defined semantics; sending a payload body on a GET request might cause some existing implementations to reject the request.

Reply:
Makes sense